### PR TITLE
instantout: add custom address to send funds to

### PR DESCRIPTION
**Pull request by sputn1ck**
This PR adds the ability to set a custom address that instantout will sweep to.

### CODE REVIEW

File: instantout/actions.go [hunk 1]
```diff
@@ -59,6 +59,7 @@ type InitInstantOutCtx struct {
 	initationHeight int32
 	outgoingChanSet loopdb.ChannelSet
 	protocolVersion ProtocolVersion
+	sweepAddress    string
```
Comment (bhandras) [inline instantout/actions.go:62]: nit: alternatively could use `*btcutil.Address` pointer so deconding can happen on the CLI.

**Review by bhandras — APPROVED** [review]
LGTM 🚀

File: instantout/actions.go [hunk 2]
```diff
@@ -165,11 +166,21 @@ func (f *FSM) InitInstantOutAction(eventCtx fsm.EventContext) fsm.EventType {
 	}
 
 	// Create the address that we'll send the funds to.
-	sweepAddress, err := f.cfg.Wallet.NextAddr(
-		f.ctx, "", walletrpc.AddressType_TAPROOT_PUBKEY, false,
-	)
-	if err != nil {
-		return f.HandleError(err)
+	var sweepAddress btcutil.Address
+	if initCtx.sweepAddress != "" {
```
Comment (hieblmi) [inline instantout/actions.go:170]: I think this should be `==`

**Review by hieblmi — COMMENTED** [review]
I think there's a small error in how we check the provided sweep address.

File: instantout/actions.go [hunk 3]
```diff
@@ -165,11 +167,17 @@ func (f *FSM) InitInstantOutAction(eventCtx fsm.EventContext) fsm.EventType {
 	}
 
 	// Create the address that we'll send the funds to.
-	sweepAddress, err := f.cfg.Wallet.NextAddr(
-		f.ctx, "", walletrpc.AddressType_TAPROOT_PUBKEY, false,
-	)
-	if err != nil {
-		return f.HandleError(err)
+	var sweepAddress btcutil.Address
```
Comment (hieblmi) [inline instantout/actions.go:170]: Ultra-mirco nit, could we simplify to

```suggestion
        var sweepAddress = initCtx.sweepAddress
	if sweepAddress == nil {
		sweepAddress, err = f.cfg.Wallet.NextAddr(
			f.ctx, lnwallet.DefaultAccountName,
			walletrpc.AddressType_TAPROOT_PUBKEY, false,
		)
		if err != nil {
			return f.HandleError(err)
		}
	}
```

**Review by hieblmi — APPROVED** [review]
great feature, LGTM

**Pull request merged**